### PR TITLE
kayvan@q2io.com - Monoids in Scala

### PR DESCRIPTION
**Pull request by kayvank**
This work is in support of the Jira ticket item: [SCALA-111](http://jira.baeldung.com/browse/SCALA-111). Monoids type class and instances for Map and List. Also, a word-frequency-counter example to test the MapMonoidInstance.

### CODE REVIEW

**Comment by kayvank** [comment]
closing pull request per instruction

File: scala-core-fp/src/test/scala/com/baeldung/scala/monoid/MapMonoidTest.scala [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package com.baeldung.scala.monoid
+
+import org.scalatest._
+
+class MapMonoidTest extends WordSpec {
```
Comment (KevinGilmore) [inline scala-core-fp/src/test/scala/com/baeldung/scala/monoid/MapMonoidTest.scala:5]: The unit test files and class names should end in "UnitTest" as per guidelines.

Comment (kayvank) [reply scala-core-fp/src/test/scala/com/baeldung/scala/monoid/MapMonoidTest.scala]: @KevinGilmore fixed.

File: scala-core-fp/src/main/scala/com/baeldung/scala/functor/Functor.scala [hunk 2]
```diff
@@ -0,0 +1,71 @@
+package com.baeldung.scala.functor
```
Comment (KevinGilmore) [inline scala-core-fp/src/main/scala/com/baeldung/scala/functor/Functor.scala:1]: I don't think the files in this package belong in the PR for this article

Comment (kayvank) [reply scala-core-fp/src/main/scala/com/baeldung/scala/functor/Functor.scala]: fixed

**Pull request merged**